From 37780648aa46a8d10ef0801d18225ae1ed859161 Mon Sep 17 00:00:00 2001
From: Miklos Vajna <vmiklos@collabora.com>
Date: Wed, 21 Jun 2023 12:06:42 +0200
Subject: [PATCH] tdf#156914 Fix --without-java build

Squashed backport of the following commits:

cui: fix --without-java build
commit af3963c32c63893949a3028396af90ee7d811e5b

cui: don't build SvxJavaClassPathDlg::SetClassPath() for the non-java case
commit 8b330bb761e05fcfabd5a30f8784046c93b91431

Change-Id: I6e6994c632c95a9662cde9ed3eddc0350dcff149
Reviewed-on: https://gerrit.libreoffice.org/c/core/+/156320
Tested-by: Jenkins
Reviewed-by: Michael Stahl <michael.stahl@allotropia.de>
---
 cui/source/options/optjava.cxx | 2 ++
 cui/source/options/optjava.hxx | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/cui/source/options/optjava.cxx b/cui/source/options/optjava.cxx
index 17f128ce9b3b..64ba41c3fe0b 100644
--- a/cui/source/options/optjava.cxx
+++ b/cui/source/options/optjava.cxx
@@ -935,6 +935,7 @@ OUString SvxJavaClassPathDlg::GetClassPath() const
     return sPath.makeStringAndClear();
 }
 
+#if HAVE_FEATURE_JAVA
 void SvxJavaClassPathDlg::SetClassPath( const OUString& _rPath )
 {
     if ( m_sOldPath.isEmpty() )
@@ -963,5 +964,6 @@ void SvxJavaClassPathDlg::SetClassPath( const OUString& _rPath )
     }
     SelectHdl_Impl(*m_xPathList);
 }
+#endif
 
 /* vim:set shiftwidth=4 softtabstop=4 expandtab: */
diff --git a/cui/source/options/optjava.hxx b/cui/source/options/optjava.hxx
index 097b7ffb5336..929dfe71738a 100644
--- a/cui/source/options/optjava.hxx
+++ b/cui/source/options/optjava.hxx
@@ -200,7 +200,9 @@ public:
     void             SetFocus() { m_xPathList->grab_focus(); }
 
     OUString                GetClassPath() const;
+#if HAVE_FEATURE_JAVA
     void                    SetClassPath( const OUString& _rPath );
+#endif
 };
 
 /* vim:set shiftwidth=4 softtabstop=4 expandtab: */
-- 
2.35.3

